index.js

```diff
@@ -65,15 +65,20 @@ function install (id) {
 /**
  * Uses webpack to build a file in memory and return the bundle.
  *
- * @param {*} file - the entry file to build.
+ * @param {*} name - library folder name
+ * @param {*} file - entry point path relative to the library folder
  * @return {Promise<string>}
  */
-function build (file) {
+function build (name, file) {
   return new Promise((resolve, reject) => {
+    const entry = requireRelative.resolve(path.join(name, file), tmp);
+    const packageJson = requireRelative.resolve(path.join(name, 'package.json'), tmp);
+    const externals = Object.keys(require(packageJson).peerDependencies || {});
```

Comment:
That's a good point. If I get some time later today I'll try to test that. I know it allows you to put both peer and dev but not sure if it gets translated into the installed package.json.

Anyways your solution is much cleaner than mine and I appreciate the PR's today 😀.